qiime2/sdk/action.py

```diff
@@ -84,6 +76,83 @@ def _run_parsl_action(action, ctx, execution_ctx, args, kwargs, inputs=[]):
         return results
 
 
+def _map_arg(arg, futures):
```

Comment:
It looks like this was also just refactored from code that was removed on line 425 - so is it then the `_is_all_proxies()` helper that's ensuring that proxies are mapped to input collections?